src/Core/Grid/Query/ContactsQueryBuilder.php

```diff
@@ -0,0 +1,133 @@
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Query;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+
+/**
+ * Class ContactsQueryBuilder is responsible for building queries for Contacts grid data.
+ */
+final class ContactsQueryBuilder extends AbstractDoctrineQueryBuilder
+{
+    /**
+     * @var DoctrineSearchCriteriaApplicatorInterface
+     */
+    private $searchCriteriaApplicator;
+
+    /**
+     * @var int
+     */
+    private $languageId;
+
+    /**
+     * @var int
+     */
+    private $shopId;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator
+     * @param int $languageId
+     * @param int $shopId
+     */
+    public function __construct(
+        Connection $connection,
+        $dbPrefix,
+        DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator,
+        $languageId,
+        $shopId
+    ) {
+        parent::__construct($connection, $dbPrefix);
+
+        $this->searchCriteriaApplicator = $searchCriteriaApplicator;
+        $this->languageId = $languageId;
+        $this->shopId = $shopId;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $qb = $this->getQueryBuilder($searchCriteria->getFilters());
+        $qb->select('c.id_contact, c.email, cl.name, cl.description');
+
+        $this->searchCriteriaApplicator
+            ->applySorting($searchCriteria, $qb)
+            ->applyPagination($searchCriteria, $qb)
+        ;
+
+        return $qb;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $qb = $this->getQueryBuilder($searchCriteria->getFilters());
+        $qb->select('COUNT(c.id_contact)');
+
+        return $qb;
+    }
+
+    /**
+     * Get generic query builder.
+     *
+     * @param array $filters
+     *
+     * @return QueryBuilder
+     */
+    private function getQueryBuilder(array $filters)
+    {
+        $qb = $this->connection
+            ->createQueryBuilder()
+            ->from($this->dbPrefix . 'contact', 'c')
+            ->innerJoin('c', $this->dbPrefix . 'contact_lang', 'cl', 'c.id_contact = cl.id_contact')
+            ->innerJoin('c', $this->dbPrefix . 'contact_shop', 'cs', 'c.id_contact = cs.id_contact')
+            ->andWhere('cl.`id_lang`=' . $this->languageId)
+            ->andWhere('cs.`id_shop`=' . $this->shopId)
```

Comment:
I'm pretty sure this will cause a problem in the "All shops" shop context. See https://github.com/PrestaShop/PrestaShop/pull/10774/files CurrencyQueryBuilder class for implementation which uses contextShopIds as array because in "All shops" there can be data displayed on the list from more then one shop.
Ofcourse I'm not sure it's the common case in all lists but it's worth to test the lists which has relations with shop entity. @sarjon wdyt?

Reply:
fixed - thanks!